silo-core/contracts/utils/liquidationHelper/LiquidationHelper.sol

```diff
@@ -0,0 +1,144 @@
+
+    /// @dev token receiver will get all rewards from liquidation, does not matter who will execute tx
+    address payable public immutable TOKENS_RECEIVER;
+
+    /// @dev address of wrapped native blockchain token eg. WETH on Ethereum
+    address public immutable NATIVE_TOKEN;
+
+    uint256 private transient _withdrawCollateral;
+    uint256 private transient _repayDebtAssets;
+
+    error NoDebtToCover();
+    error STokenNotSupported();
+
+    /// @param _nativeToken address of wrapped native blockchain token eg. WETH on Ethereum
+    /// @param _exchangeProxy exchange address, where to send swap data on liquidation
+    /// @param _tokensReceiver all leftover tokens (debt and collateral) will be send to this address after liquidation
+    constructor (
+        address _nativeToken,
+        address _exchangeProxy,
+        address payable _tokensReceiver
+    ) DexSwap(_exchangeProxy) {
+        NATIVE_TOKEN = _nativeToken;
+        EXCHANGE_PROXY = _exchangeProxy;
+        TOKENS_RECEIVER = _tokensReceiver;
+    }
+
+    /// @inheritdoc ILiquidationHelper
+    /// @dev entry point for liquidation
+    /// @notice for now we does not support liquidation with sTokens
+    /// on profitable liquidation we will revert because we will not be able to repay flashloan with fee
+    function executeLiquidation(
+        ISilo _flashLoanFrom,
+        address _debtAsset,
+        uint256 _maxDebtToCover,
+        LiquidationData calldata _liquidation,
+        DexSwapInput[] calldata _swapsInputs0x
+    ) external returns (uint256 withdrawCollateral, uint256 repayDebtAssets) {
+        require(_maxDebtToCover != 0, NoDebtToCover());
+
+        _flashLoanFrom.flashLoan(this, _debtAsset, _maxDebtToCover, abi.encode(_liquidation, _swapsInputs0x));
+        IERC20(_debtAsset).approve(address(_flashLoanFrom), 0);
+
+        withdrawCollateral = _withdrawCollateral;
+        repayDebtAssets = _repayDebtAssets;
+    }
+
+    /// @inheritdoc IERC3156FlashBorrower
+    function onFlashLoan(
+        address /* _initiator */,
+        address _debtAsset,
+        uint256 _maxDebtToCover,
+        uint256 _fee,
+        bytes calldata _data
+    )
+        external
+        returns (bytes32)
+    {
+        (
+            LiquidationData memory _liquidation,
+            DexSwapInput[] memory _swapInputs
+        ) = abi.decode(_data, (LiquidationData, DexSwapInput[]));
+
+        IERC20(_debtAsset).approve(address(_liquidation.hook), _maxDebtToCover);
+
+        bool receiveSToken;
+
+        (
+            _withdrawCollateral, _repayDebtAssets
+        ) = _liquidation.hook.liquidationCall(
+            _liquidation.collateralAsset, _debtAsset, _liquidation.user, _maxDebtToCover, receiveSToken
+        );
+
+        IERC20(_debtAsset).approve(address(_liquidation.hook), 0);
+
+        if (_liquidation.collateralAsset == _debtAsset) {
+            uint256 balance = IERC20(_liquidation.collateralAsset).balanceOf(address(this));
```

Comment:
We need the balance of the collateralAsset in both execution paths (in the `if` block and int the `else` block). Maybe we can move it before `if` and use it in both blocks?
```suggestion
          uint256 balance = IERC20(_liquidation.collateralAsset).balanceOf(address(this));

        if (_liquidation.collateralAsset == _debtAsset) {

```

Reply:
we can not, because balance can change eg because of swap.